dotcom-rendering/src/web/components/Liveness.importable.tsx

```diff
@@ -0,0 +1,186 @@
+import { css } from '@emotion/react';
+import { useEffect, useState } from 'react';
+import { useApi } from '../lib/useApi';
+
+type Props = {
+	pageId: string;
+	webTitle: string;
+	ajaxUrl: string;
+};
+
+// TODO: Break this out into its own component
+const Toast = ({
+	onClick,
+	noOfNewPosts,
+}: {
+	onClick: () => void;
+	noOfNewPosts: number;
+}) => {
+	// TODO: Style and absolute position this component
+	return (
+		<nav
+			css={css`
+				position: fixed;
+				top: 20px;
+				left: 50px;
+				display: flex;
+				width: 100%;
+				align-items: center;
+			`}
+		>
+			<button onClick={onClick}>{`${noOfNewPosts} blah`}</button>
+		</nav>
+	);
+};
+
+const isServer = typeof window === 'undefined';
+
+function hydrateBlocks() {
+	console.log('hydrateBlocks');
+	// TODO Call existing hydration solution targetted to these elements
+	// Maybe we can make doHydration work for all gu-islands things
+	// where the gu-hydrated data attribute isn't set?
+}
+
+/**
+ * insertNewBlocks takes html and inserts it at the top of the liveblog
+ *
+ * @param {string} html The block html to be inserted
+ * @returns void
+ */
+function insertNewBlocks(html: string) {
+	const latestBlock = document.querySelector('#maincontent :first-child');
+	if (!latestBlock) return;
+	latestBlock.insertAdjacentHTML('beforebegin', html);
+}
+
+/**
+ * revealNewBlocks - style any blocks that have been inserted but are hidden such that
+ * they are revealed
+ */
+function revealNewBlocks() {
+	console.log('revealNewBlocks');
+}
+
+/**
+ * This abstracts the code to construct the url we use for polling. Of note is the fact
+ * we have put `latestBlockId` in react state; by doing this we are able to change the url
+ * that SWR uses for the request whilst still benefitting from the polling logic
+ *
+ * @returns string
+ */
+function getKey(
+	pageId: string,
+	ajaxUrl: string,
+	filterKeyEvents: boolean,
+	latestBlockId: string,
+): string {
+	// Construct the url to poll
+	const url = new URL(`${pageId}.json`, ajaxUrl);
+	url.searchParams.set('lastUpdate', latestBlockId);
+	url.searchParams.set('isLivePage', 'true');
```

Comment:
Not sure we need to set this at all - but we should only poll when the liveblog is live (rather than dead)

Reply:
I think we can simply not send this query param

Reply:
I'm wrong. We do need to send this param. If [we don't](https://api.nextgen.guardianapps.co.uk/world/live/2022/feb/01/covid-live-news-denmark-lifts-all-covid-restrictions-despite-record-cases-liz-truss-tests-positive.json?lastUpdate=block-61f98fb28f08ed69ff280052&filterKeyEvents=false) then the response no longer contains the html.

This is how Frontend works. Arguably, DCR does not have to follow this pattern but it might be better to keep things consistent between the two apis.

Reply:
To your other point regarding when we poll. The code here in this file is only loaded on live blogs so we would never start polling unless the blog is live on page load.

This does raise the question of how do we _stop_ polling? Frontend uses a backoff strategy but the code here is set to poll for as long as the tab is open. Maybe we can include the blog status in the response and use that to stop polling?